lib/logger.js

```diff
@@ -0,0 +1,80 @@
+const winston = require('winston')
+const { format, transports } = winston
+const { combine, timestamp, label, printf } = format
+
+// Usage examples:
+// Simple logger setup with module name and default config:
+//   const logger = require('./logger.js').setup('zwc')
+// Logger setup with module name and specific config:
+//   const logger = require('./logger.js').setup('zwc',
+//     { enabled: true, level: 'info', logToFile: true, filename: "zwavejs2mqtt.log" }
+//   )
+
+// Make sure every config value is defined
+const sanitizedConfig = (module, config) => {
+  config = config || {}
+  return {
+    module: module !== undefined ? module : '-',
+    enabled: config.enabled !== undefined ? config.enabled : true,
+    level: config.level !== undefined ? config.level : 'info',
+    logToFile: config.logToFile !== undefined ? config.logToFile : false,
+    filename:
+      config.filename !== undefined ? config.filename : 'zwavejs2mqtt.log'
+  }
+}
+
+// Format loglevel upper-case:
+const upperCaseLevel = format(info => {
+  info.level = info.level.toUpperCase()
+  return info
+})
+
+// Custom combined logging format:
+const customFormat = config =>
```

Comment:
Does this handles errors? I mean if I log an error is the stack trace printed to log output?

Reply:
I guess something like `errors({ stack: true })` will have to be added to the format as seen here:
<https://stackoverflow.com/questions/47231677/how-to-log-full-stack-trace-with-winston-3>
I'll add it.

Reply:
@robertsLando It's done and I've added 3 example usages in ZwaveClient.js.
The 2nd `error` parameter to the log statements may be removed again, if we don't require a stack trace for these logs.